scripts/projectValidation/projectSchema.ts

```diff
@@ -0,0 +1,120 @@
+import { z, ZodTypeAny } from "zod";
+
+type EnumArrayType = [string, ...string[]];
+
+export const categories = [
+  "defi",
+  "dex",
+  "bridges",
+  "lending",
+  "nfts",
+  "gaming",
+  "social",
+  "nfts",
+  "wallets",
+  "dao",
+  "other",
+];
+
+const safeImageExtensions = [".jpg", ".jpeg", ".png", ".gif", ".bmp", ".webp"];
+
+const safeImageMimeTypes = [
+  "image/jpeg", // For both .jpg and .jpeg
+  "image/png",
+  "image/gif",
+  "image/bmp",
+  "image/webp",
+];
+
+const idSchema = z
+  .string()
+  .regex(/^[a-zA-Z0-9-_.]+$/)
+  .min(2)
+  .max(100);
+
+const chainEnumSchema = z.enum(["moonriver", "moonbeam"]);
+
+const defaultStringSChema = z.string().min(2).max(100);
+
+const imageSchema = z.object({
+  fileName: z.union(
```

Comment:
Do you want to be a bit more specific? Like line length and without special characters?